### ISSUE TITLE
is plugin ignoring method references? 

### ISSUE THREAD

**Issue by 62mkv** [Stefku/intellij-reference-diagram#76]
I've noticed that method reference diagram is not considering method names used with "method reference syntax". This is what I get for such class: 

```java
public class DummyService {
    public Mono<String> getStringWithMethodReference() {
        return Mono.just("Hello, world")
                .map(this::processString);
    }
    public Mono<String> getStringWithLambda() {
        return Mono.just("Hello, world")
                .map(s -> processString(s));
    }
    private String processString(String source) {
        return source.toUpperCase(Locale.ROOT);
    }
}
```

![image](https://user-images.githubusercontent.com/5255127/154032331-6ed33411-d06e-4130-a69b-e9b89ab1a23b.png)

PS: by the way, I would like to be able to contribute a fix for this, so if there're any hints/suggestions, please share! I did not yet take a look at the code though..

**Comment by Stefku** [comment]
Hi @62mkv, you probably found a bug. Thanks for reporting.

You are very welcome to contribute. I think a good point to start is ReferenceDiagramDataModel. The constructor gets called with a PsiClass where the user wanted to render the diagram. Give it a try and come back to me if you struggle.

**Comment by 62mkv** [comment]
Hi @Stefku only having cloned the project, I realized that it's purely for *Ultimate* which unfortunately means I can not contribute to this awesome project, as I only have IC edition at home! Will buy you coffee instead!

**Comment by Stefku** [comment]
Yes, since JetBrains has licensed the diagram library, the plugin is just for Ultimate version. I would know a good and easy to use possibility of a free open source library we could change it (see also #56).

Thanks for the coffee @62mkv! I really appreciate it! I'm glad, that other people also have the use for this plugin.
